crates/bevy_core_pipeline/src/prepass/mod.rs

```diff
@@ -0,0 +1,147 @@
+//! Run a prepass before the main pass to get the depth and/or normals texture.
+//! The depth prepass texture is then used by the main pass to reduce overdraw.
+//!
+//! To enable the prepass, you need to add a `PrepassSettings` component to a `Camera`.
+//! Both textures are available on the `PrepassTextures` component attached to each `Camera` with a `PrepassSettings`
+//!
+//! Currently only works for 3d
+
+pub mod node;
+
+use bevy_ecs::prelude::*;
+use bevy_render::{
+    render_phase::{CachedRenderPipelinePhaseItem, DrawFunctionId, EntityPhaseItem, PhaseItem},
+    render_resource::{CachedRenderPipelineId, Extent3d},
+    texture::CachedTexture,
+};
+use bevy_utils::FloatOrd;
+
+/// Add a [`PrepassSettings`] component to a [`crate::prelude::Camera3d`] to perform a depth and/or normal prepass.
+/// These textures are useful for various screen-space effects and reducing overdraw in the main pass.
+///
+/// The textures are automatically added to the default mesh view bindings. You can also get the raw textures
+/// by querying the [`ViewPrepassTextures`] component on the camera with the [`PrepassSettings`].
+///
+/// When using the default mesh view bindings you should be able to use `prepass_depth()`
+/// and `prepass_normal()` to load the related textures. These functions are defined in `bevy_pbr::utils`.
+/// See the `shader_prepass` example that shows how to use it.
+#[derive(Clone, Component)]
+pub struct PrepassSettings {
+    /// If true then depth values will be copied to a separate texture available to the main pass.
+    /// This will help reduce overdraw in the main pass.
+    pub depth_enabled: bool,
+    /// If true then vertex world normals will be copied to a separate texture available to the main pass.
+    pub normal_enabled: bool,
+}
+
+impl Default for PrepassSettings {
+    fn default() -> Self {
+        Self {
+            depth_enabled: true,
+            normal_enabled: false,
+        }
+    }
+}
+
+/// Textures that are written to by the prepass.
+///
+/// The texture only exists if the relevant option on [`PrepassSettings`] is `true`.
+#[derive(Component)]
+pub struct ViewPrepassTextures {
```

Comment:
Rename to just `PrepassTextures`?

Reply:
If they are per-view then it should be ViewPrepassTextures. And I would expect them to be per-view.

Reply:
They are per-view so I'll leave it as is.